sasmodels/kernelpy.py

```diff
@@ -309,6 +324,22 @@ def vector_Iq(q, *args):
         vector_Iq.vectorized = True
         model_info.Iq = vector_Iq
 
+def _create_vector_Fq(model_info):
+    """
+    Define Fq as a vector function if it exists.
+    """
+    # Note that this is doing slightly too much work since we are composing
+    # two vector results which are then interlaced via _pack_fq above.
+    Fq = model_info.Fq
+    if callable(Fq) and not getattr(Fq, 'vectorized', False):
+        def vector_Fq(q, *args):
+            """
+            Vectorized 1D kernel returning Fq, Fq**2
+            """
+            fq, fqsq = zip(*(Fq(qi, *args) for qi in q))
+            return np.array(fq), np.array(fqsq)
```

Comment:
I'm not certain, but slicing a numpy array might be considerably faster than `zip(*)`

Reply:
Surprisingly, the zip version is faster:
```
In [15]: %timeit x,y = [np.array(v) for v in zip(*(((i, i) for i in np.arange(1000))))]
242 μs ± 7.15 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [16]: %timeit x,y = (a:=np.array(list((i, i) for i in np.arange(1000))))[:,0],a[:,1]
389 μs ± 4.75 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```
But in any case, this cost is small compared to evaluating the function at each q.